### ISSUE TITLE
Adding a background component to the topbar disables right and left buttons

### ISSUE THREAD

**Issue by alan-nakagiri** [wix/react-native-navigation#5995]
### Issue Description

The main issue is when a background component is added to the top bar the right and left buttons do not work. This issue appears to have been introduced after v3.7.0. 

For some context I am trying to achieve placing a progress meter in the topbar that appears as below:

![image](https://user-images.githubusercontent.com/11300359/75848205-59b91200-5d96-11ea-98a8-4908956fc59f.png)

### Steps to Reproduce / Code Snippets / Screenshots
- set the root to screen with a topbar component (iOS) or background component (android) 
- push to new screen that has a back arrow enabled
- the back arrow should not be tappable

To get this progress bar rendering on iOS and Android I've had to implement different solutions on iOS vs Android on v3.7.0 I am unsure if this is expected but I thought it was worth pointing out.

iOS
- Setting the root with is the only way to set a component in the topbar
- Pushing a new screen with the topbar component does not add a component to the topbar
- Adding the component is a key of topbar `topBar: { component: { name: 'compName' } }` 

Android
- Setting the root with the topBar background component is also required to display a component.
- Pushing a new screen with a topbar background component is required to keep the component there.
- Adding the component is nested in topbar under background. `topBar: { background: { component: { name: 'compName' }  }  }`

---
### Environment
* React Native Navigation version: v4.8 (also tested in playground on v6.0.1)
* React Native version: 0.60.1
* Platform(s) (iOS, Android, or both?): Both
* Device info (Simulator/Device? OS version? Debug/Release?): Simulator, Debug

**Comment by alan-nakagiri** [comment]
@guyca Just following up Do you need any more information regarding this issue?

**Comment by bneigher** [comment]
Why is there a different implementation for iOS and Android?

**Comment by alan-nakagiri** [comment]
@bneigher I had a different implementation for iOS and Android because they behave differently. 

`topBar: { background: { component: { name: 'compName' } } }` works on Android, but on iOS adding a background component that way somehow disables any buttons on the topBar.

**Comment by bneigher** [comment]
@alan-nakagiri - oh sorry, that was directed toward the wix team.  I believe there were some unification of the api for some other things lately, maybe this is one that's slipped under their radar.

BTW, on the latest version (6.4.0) does this work anymore on iOS? I'm seemingly unable to set a background for the whole top bar, only able to do so with the title area.

**Comment by alan-nakagiri** [comment]
@bneigher Ah gotcha it doesn't sound like this is fixed yet. I last tried upgrading to 6.0.1 but then realized we had to revert back to 3.7.0.

And using 3.7.0 I was able only able to set the background component to the topBar by calling `setRoot` with the following config nested under `topBar`.

```
topBar: {
      background: {
        color:  topBarColor,
        component: isIOS() ? {} : { name: screens.MODULE_HEADER },
      },
      component: isIOS() ? { name: screens.MODULE_HEADER } : {},
}
```

and this when we call push
```
topBar: {
      background: {
        color:  topBarColor,
        component: isIOS() ? {} : { name: screens.MODULE_HEADER },
      },
}
```

We have yet to successfully set the background of the topBar via push without setting a root with it first. It's been a handcuff for sure, but we've been able to manage for now.

I was hoping to hear something from the Wix team before trying again, but it seems doubtful the issue has been resolved.
